server/api/account.js

```diff
@@ -82,4 +106,105 @@ module.exports.init = function(server) {
     server.post('/api/account/login', passport.authenticate('local'), function(req, res) {
         res.send({ success: true, user: req.user, token: jwt.sign(req.user, config.secret) });
     });
+
+    server.post('/api/account/password-reset-finish', function(req, res) {
+        var responseSent = false;
+        var resetUser;
+
+        if(!req.body.id || !req.body.token || !req.body.newPassword) {
+            return res.send({ success: false, message: 'Invalid parameters' });
+        }
+
+        userRepository.getUserById(req.body.id).then(user => {
+            if(!user) {
+                throw new Error('User id not found ' + req.body.id);
+            }
+
+            if(!user.resetToken) {
+                throw new Error('Got unexpected reset request for user ' + user.username);
+            }
+
+            resetUser = user;
```

Comment:
I'm kind of iffy on setting external variables like this vs explicitly returning the data and piping it into the next `then` step. But it's probably fine? Might just be a purity thing on my part.